### PR TITLE
Support for 2016 style, fixup.js, and experimental styles

### PR DESCRIPTION
**Pull request by marcoscaceres**

### CODE REVIEW

**Comment by marcoscaceres** [comment]
@tobie, can you review file js/core/override-configuration.js:
https://github.com/w3c/respec/commit/2333164c36e857315eae693f51133dd342304800

I only made cosmetic changes to the tests, so they don't need review.

**Comment by marcoscaceres** [comment]
@plehegar, @tripu, @halindrome please review: 
https://github.com/w3c/respec/commit/cceb8ada6e96c9f64dcad7f40d16e1802beb8ffa

**Comment by halindrome** [comment]
Crazy question - but is there a way to get a version of the new styles for "unofficial drafts"?  I get that CG is not supported (yet), but UDs are things that are usually being developed as part of some proposal / strawman process and it would be helpful if they looked the same.

**Comment by marcoscaceres** [comment]
UD should work... though not sure what they look like yet as I haven't tried.

**Comment by halindrome** [comment]
Huh.  I will test it.

**Comment by marcoscaceres** [comment]
@halindrome, would really appreciate a review of this stuff tho. After this, I got a complete rewrite of the ReSpec test suite coming using Jasmine 2.1. Pretty keen to land as it's blocking that.

**Comment by halindrome** [comment]
Understood.  I have verified that the UD stuff works btw.  I will do a review - but I have been in web payments standards meetings all day and it is 11:30 pm - I am going to get a few hours of shut eye first.

**Comment by marcoscaceres** [comment]
Perfect! Meetings are the best time to do code reviews! :dancer: :smile_cat:

**Comment by halindrome** [comment]
Accurate.

On Mon, Feb 22, 2016 at 11:32 PM, Marcos Caceres <notifications@github.com>
wrote:

> Perfect! Meetings are the best time to do code reviews! [image: :dancer:] [image:
> :smile_cat:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/w3c/respec/pull/571#issuecomment-187582977>.
>



-- 
Shane McCarron
halindrome@gmail.com

**Comment by marcoscaceres** [comment]
@tobie, @halindrome, thank you for taking the time to review this. :trophy: :beers: for you both!

**Comment by plehegar** [comment]
:+1:
seems to work for me.

**Comment by plehegar** [comment]
I don't mind it being smarter, but the pubrules checker will mandate it at the end.

**Comment by halindrome** [comment]
Really? Ok.

**Comment by plehegar** [comment]
btw, it looks like we're going to need to revise the pubrules again in a month, to deal with https. If that happens and fixup.js gets modify to use DOMContentLoaded, we could at that time relax the position of fixup.js in the markup.

File: js/w3c/style.js [hunk 1]
```diff
@@ -1,55 +1,115 @@
...
-                cb();
-            }
+    function(utils) {
+      function attachFixupScript(doc, version){
+        var script = doc.createElement("script");
+        script.async = "async";
+        var helperScript = "https://www.w3.org/scripts/TR/{version}/fixup.js"
+          .replace("{version}", version);
+        script.src = helperScript;
+        doc.head.appendChild(script);
```
Comment (halindrome) [inline js/w3c/style.js]: Can we please change this to doc. body.appendChild so that pubrules will pass?  Then LGTM.

**Comment by halindrome** [comment]
Note that we should NOT deploy this prior to 1 March or it will break auto-publications.  @plehegar pointed this out in irc.w3.org#pub today

**Comment by marcoscaceres** [comment]
No probs about holding off till the 1st of March.

**Comment by marcoscaceres** [comment]
Please don't release this until after we have checked on more real specs.

**Comment by halindrome** [comment]
I am not sure what that means.  If the new styles are live 1 March then
they are live.  These changes don't effect anything except which
stylesheets are loaded by default (and the ability to automatically support
a 2017 sheet when it starts to become available.

On Mon, Feb 29, 2016 at 5:56 PM, Marcos Caceres <notifications@github.com>
wrote:

> Please don't release this until after we have checked on more real specs.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/w3c/respec/pull/571#issuecomment-190455228>.
>



-- 
Shane McCarron
halindrome@gmail.com

**Comment by tripu** [comment]
We'll update [the version of spec-generator that Echidna uses](https://labs.w3.org/spec-generator/) to pull that new release of `respec` when it's published.

**Pull request merged**